### PR TITLE
refactor: clean up logger handling, #68

### PR DESCRIPTION
**Pull request by xavierchow**
This addresses #68,
- use `bunyan-logger-factory` to initialize the logger instance.
- fix logger gyp compiling issue under node v10 with the usage of `bunyan-logger-factory`
- set up logging middleware in the `server/boot/register-logger.js` instead of spreading middlewares and handling everywhere.

### CODE REVIEW

**Comment by makara** [comment]
Did you also remove a few loggers like debug and ringbuffer, and the middlewares of tracelog and errorhandler? What's the plan?

**Comment by xavierchow** [comment]
- debug
  yes, `debug` should be strictly distinguished from log, if you need log output just set stream as stdout.
- ringbuffer
 yes, it's not really used in prod, to test the log itself I use `file` like this: https://github.com/starbucks-china/order-management-system/blob/1d3f196bf2641ac6460f1e60e2b30c1b3ad36b39/server/test/logger.test.js#L20-L30

- tracelog
 Auto-generating the `x-transaction-id` is not good, it's a mandatory header which should be passed through upstream unless it's Gateway like service.  cc @zbal 

- errorhandler
 Moved to `server/boot/register-logger.js`

**Comment by xavierchow** [comment]
Creating #73 to follow up.

File: server/boot/lib-model.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+'use strict';
+
+const loggerFactory = require('bunyan-logger-factory');
+
+module.exports = app => {
+  app.lib = app.lib || {};
+
+  app.logger = app.lib.logger = loggerFactory.init({
+    logName: 'wcl-service',
```
Comment (haishanh) [inline server/boot/lib-model.js]: I think it's important for different services to use a different log name, i.e. should use `app.get(logName')`

Comment (xavierchow) [reply server/boot/lib-model.js]: good catch

**Pull request merged**